### PR TITLE
Support for reading Space Ranger V2 with read10xVisium()

### PR DESCRIPTION
**Pull request by HelenaLC**
related to issue #108

### CODE REVIEW

**Review by drighelli — APPROVED** [review]
seems a good patch to me!
(nice touch on the "header" argument!) :)

**Comment by HelenaLC** [comment]
@lmweber Any idea what's up with all the check errors? Anything to worry about? Did we have these before?

**Comment by drighelli** [comment]
@HelenaLC it fails in the vignette building

```
Quitting from lines 64-67 (SpatialExperiment.Rmd) 
Error: Error: processing vignette 'SpatialExperiment.Rmd' failed with diagnostics:
'names' attribute [2] must be the same length as the vector [1]
--- failed re-building ‘SpatialExperiment.Rmd’
SUMMARY: processing the following file failed:
  ‘SpatialExperiment.Rmd’
```

File: R/read10xVisium.R
Comment (drighelli) [inline R/read10xVisium.R:130]: @HelenaLC could the error (for the vignette) be here?

**Comment by HelenaLC** [comment]
Indeed, sorry about that. There was a bug in the file path construction for multiple samples; should be fixed now.

**Comment by lmweber** [comment]
This looks great @HelenaLC , thanks! The `sprintf` / `file.exists` and `!grepl` solutions are really concise and clean.

I think we might have a merge conflict with the version bump from the last Bioc release date. I'll check that in my clone of the repo and will push any version bump fixes here if we need them before merging the branch.

**Comment by lmweber** [comment]
I think we can also update this part of the docs to use the new `tissue_positions.csv` filename:
```
#' sample \cr
#' · | — outs \cr
#' · · | — raw/filtered_feature_bc_matrix.h5 \cr
#' · · | — raw/filtered_feature_bc_matrix    \cr
#' · · · | — barcodes.tsv.gz \cr
#' · · · | — features.tsv.gz \cr
#' · · · | — matrix.mtx.gz   \cr
#' · · | — spatial \cr
#' · · · | — scalefactors_json.json    \cr
#' · · · | — tissue_lowres_image.png   \cr
#' · · · | — tissue_positions_list.csv \cr
```

**Comment by lmweber** [comment]
I added a commit above to update the `tissue_positions.csv` filename in the docs and vignette.

I think we can merge this one once the checks finish, and then the easiest way to update the version number merge conflict will be if I make a new pull request with the merge conflict commit. (I'll also use that pull request to update the citation for our paper in DESCRIPTION, which has also been on the to do list.)

**Pull request merged**
